core/domain/topic_services.py

```diff
@@ -940,6 +940,25 @@ def get_topic_rights(topic_id, strict=True):
     return get_topic_rights_from_model(model)
 
 
+def get_multi_topic_rights(topic_ids):
+    """Returns the rights of all topics whose topic ids are passed in.
+
+    Args:
+        topic_ids: list(str). The IDs of topics for which rights are to be
+            returned.
+
+    Returns:
+        list(TopicRights). The list of rights of all given topics present in
+            the datastore.
+    """
+    topic_rights_models = topic_models.TopicRightsModel.get_multi(topic_ids)
+    topic_rights = [
```

Comment:
Don't elide topic rights that are None. Include them and handle them too (and update the docstring Returns value accordingly).

Otherwise your index calculation in the controller that uses this will be wrong.

Reply:
I thought of this, but the `get_multi_topic_summaries` function also omits invalid topics. So, shoudn't both these match in the end?

Reply:
Ah, I see.

I think get_multi's (by ID) should always return None for cases where stuff is not found. We do that elsewhere throughout the codebase, so let's not break that convention.

Reply:
In the topic_services function, the `get_multi` call to the gae_model does return None for invalid topics, but then they are filtered out before calling `get_<entity>_from_model`. So, here, instead of filtering out, should I just set None in the final list?

Reply:
We may need to do tidying up in the controller layer, but for any get_multi calls to the domain layer, the list that's returned should include None values for missing entities.

Reply:
Ok, done.